### ISSUE TITLE
Creating unit test for fine-tuning

### ISSUE THREAD

**Issue by WenkelF** [WenkelF/graphium#4]
The unit test should check for
- Correct architectural modifications between pretrained and fine-tuning model
- Correct (un)freezing in the fine-tuning training

**Comment by WenkelF** [comment]
Check following:
- Gradients enabled/disabled for (un)frozen layers
- Continuous training curve when stopping & restarting training
- Context switcheng between IPU/GPU
